float8_experimental/float8_tensor.py

```diff
@@ -14,6 +14,19 @@
 aten = torch.ops.aten
 
 
+def tensor_already_casted_to_fp8(tensor: torch.Tensor) -> bool:
+    """
+    Check if the tensor is already casted to fp8
+    """
+    if isinstance(tensor, Float8Tensor):
+        return True
+    elif isinstance(tensor, DTensor) and isinstance(tensor._local_tensor, Float8Tensor):
+        # TODO: shall we stick to public API and directly use tensor.to_local() here?
```

Comment:
I wonder if in general, subclasses composing with other subclasses should have a generic way to determine the nested types

Reply:
Yeah, I'll also need to think about how this should behave, this seems in general how we can get subclass hierarchy from some type info, simply trying out `type.mro` or `inspect.getmro` seems not working for me. Maybe need to check with @ezyang @bdhirsh to see if there's any better suggestions